src/domain/auth/login.use-case.ts

```diff
@@ -16,6 +16,32 @@ interface LoginUseCaseResponse {
   token: string;
 }
 
+export class LoginUseCase {
+  constructor(private readonly repository = AppDataSource.getRepository(User)) {}
```

Comment:
nesse caso pode ser uma propriedade intrinseca à classe, justamente por vc nao querer que seja sobreescrevida quando o use case for instanciado, certo?
Ai pode ate remover o constructor

```suggestion
  private readonly repository = AppDataSource.getRepository(User)
```